### PR TITLE
Use spans instead of single locations in AST

### LINKED ISSUE #121: Use Spans in the AST
This will mean that we can get rid of the `LExpr::span` and `LPattern::span` methods, making our locations much less error prone. The removal of those methods will also make it easier to switch to byte positions in #108, and make the switch to LALRPOP easier, as discussed in #69.

I have started this in #116, but it seemed to be more than I could chew at this stage.

### PR DESCRIPTION
**Pull request by Marwes**
Continuation of #116 

Closes #121

### CODE REVIEW

File: parser/src/lib.rs [hunk 1]
```diff
@@ -401,16 +398,16 @@ impl<'s, I, Id, F> ParserEnv<I, F>
...
-            located(Location {
-                        column: CharPos(position.column as usize),
-                        line: position.line as u32,
-                        absolute: BytePos(0),
-                    },
-                    expr)
+            pos::spanned(Span {
+                             start: position,
+                             // ARGH - TODO
+                             end: position,
```
Comment (Marwes) [inline parser/src/lib.rs]: Haven't fixed this yet so spans are not working yet

**Comment by brendanzab** [comment]
Might want to rebase onto master so that beta passes as well

**Comment by Marwes** [comment]
Oh yeah, forgot that.

**Comment by brendanzab** [comment]
Hmm - did you rebase onto the right master branch?

**Comment by Marwes** [comment]
Now it should be fixed, accidentally had the addition of stable commit to my local master and I forgot to pull in the master from github.

**Pull request merged**